src/main/scala/inox/parser/elaboration/Constraints.scala

```diff
@@ -143,6 +152,19 @@ trait Constraints { self: IRs with SimpleTypes with ElaborationErrors =>
     case x => Eventual.pure(x)
   }
 
+  implicit lazy val typeSequenceUnifiable: Unifiable[Seq[SimpleTypes.Type]] = Unifiable {
+    a: Seq[SimpleTypes.Type] => for {
+      ss <- Eventual.sequence(a.map(tpe => Eventual.withUnifier { implicit unifier => unifier(tpe) }))
+    } yield ss
+  }
+
+  implicit lazy val oneOfValueUnifiable: Unifiable[(SimpleTypes.Type, Seq[SimpleTypes.Type])] = Unifiable {
```

Comment:
It is used then we would like to unify values inside the `typeOptionsMap` since this map contains the specified pair as a value. I agree that `pairUnifiable` is more general.